### PR TITLE
[#335] Forgot password popup

### PR DESCRIPTION
**Pull request by Aekal**
- Added forgot password popup with email validation
- Added account icon to header
- Moved props from SignUp to vuex (after closing popup, login is set to default)

#335 #355 #316 

zrzut ekranu 2017-12-04 o 10 35 27

zrzut ekranu 2017-12-04 o 10 35 07

zrzut ekranu 2017-12-04 o 10 35 21

### CODE REVIEW

File: src/components/core/blocks/Auth/SignUp.vue [hunk 1]
```diff
@@ -3,19 +3,23 @@
...
 import { mapState } from 'vuex'
 
 export default {
   name: 'sign-up',
   components: {
     Login,
-    Register
+    Register,
+    ForgotPass,
+    mapState
```
Comment (filrak) [inline src/components/core/blocks/Auth/SignUp.vue]: mapState isn't component

File: src/themes/default/components/core/blocks/Auth/ForgotPass.vue [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  },
+  mixins: [coreComponent('core/blocks/Auth/ForgotPass')],
+  components: {
+    ButtonFull
+  }
+}
+</script>
+
+<style lang="scss" scoped>
+  @import '../../../../css/text.scss';
```
Comment (filrak) [inline src/themes/default/components/core/blocks/Auth/ForgotPass.vue]: same as in https://github.com/DivanteLtd/vue-storefront/pull/365

File: src/themes/default/components/core/blocks/Header/AccountIcon.vue [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+<script>
+import { coreComponent } from 'lib/themes'
+
+export default {
+  mixins: [coreComponent('core/blocks/Header/AccountIcon')]
+}
+</script>
+
+<style lang="scss" scoped>
```
Comment (filrak) [inline src/themes/default/components/core/blocks/Header/AccountIcon.vue]: if its unused please delete it ;)

File: src/themes/default/components/core/blocks/Header/Header.vue [hunk 4]
```diff
@@ -108,6 +111,9 @@ export default {
         .wishlist-icon {
```
Comment (filrak) [inline src/themes/default/components/core/blocks/Header/Header.vue]: for all this icons add one proper class to visibility.scss

File: src/components/core/blocks/Auth/SignUp.vue (hunk 1 above)
Comment (Aekal) [reply src/components/core/blocks/Auth/SignUp.vue]: Done

File: src/themes/default/components/core/blocks/Auth/ForgotPass.vue (hunk 2 above)
Comment (Aekal) [reply src/themes/default/components/core/blocks/Auth/ForgotPass.vue]: Done

File: src/themes/default/components/core/blocks/Header/AccountIcon.vue (hunk 3 above)
Comment (Aekal) [reply src/themes/default/components/core/blocks/Header/AccountIcon.vue]: Deleted in few components

File: src/themes/default/components/core/blocks/Header/Header.vue (hunk 4 above)
Comment (Aekal) [reply src/themes/default/components/core/blocks/Header/Header.vue]: I added hidden-xs class to icons

**Comment by filrak** [comment]
Thank you! ;)

**Pull request merged**